src/libawkward/array/UnionArray.cpp

```diff
@@ -516,7 +516,11 @@ namespace awkward {
 
   template <typename T, typename I>
   const std::shared_ptr<Content> UnionArrayOf<T, I>::flatten(int64_t axis) const {
-    throw std::runtime_error("FIXME: UnionArray::flatten");
+    std::vector<std::shared_ptr<Content>> contents;
+    for (auto content : contents_) {
+      contents.emplace_back(content.get()->flatten(axis));
+    }
+    return std::make_shared<UnionArrayOf<T, I>>(identities_, parameters_, tags_, index_, contents);
```

Comment:
Same as for RecordArray: if the `axis` is such that the lengths of each `content` do not change, all is well (the `index_` is still valid). But if any of those lengths change, then the meaning of the UnionArray would be garbled.

Reply:
addressed in #102